File: hyein/week10/입국심사.py
def solution(n, times):
    answer = 0
    left = min(times)
    right = max(times)*n

    while left <= right:
        mid = (left+ right) // 2
        checked = 0
        for time in times:
            checked += mid // time
            if checked >= n:
                break
        
        if checked >= n:
            answer = mid
            right = mid - 1
        elif checked < n:
            left = mid + 1
            
    return answer

File: hyejee/week10/입국심사.swift
import Foundation

func solution(_ n:Int, _ times:[Int]) -> Int64 {
    var answer: Int64 = 0
    let sortedTimes: [Int] = times.sorted()
    var lowerBound: Int64 = 0
    var upperBound: Int64 = Int64(n * sortedTimes[0])
    
    while lowerBound <= upperBound {
        let middle: Int64 = (lowerBound + upperBound) / 2
        
        var maxCount: Int64 = 0
        
        for time in sortedTimes {
            maxCount += middle / Int64(time)
        } 
        
        if maxCount < n {
            lowerBound = middle + 1
        }
        else {
            answer = middle
            upperBound = middle - 1
        }
    }
    
    return answer
}
